music21/metadata/__init__.py

```diff
@@ -96,161 +184,499 @@ class Metadata(base.Music21Object):
-    # !!!OMD: Movement designation or movement name.
-    # !!!OPS: Opus number.
-    # !!!ONM: Number.
-    # !!!OVM: Volume.
-    # !!!ODE: Dedication.
-    # !!!OCO: Commission
-    # !!!GTL: Group Title.
-    # !!!GAW: Associated Work.
-    # !!!GCO: Collection designation.
-    # !!!TXO: Original language of vocal/choral text.
-    # !!!TXL: Language of the encoded vocal/choral text.
-    # !!!OCY: Country of composition.
-    # !!!OPC: City, town or village of composition.
-
-    workIdAbbreviationDict = {
-        'gaw': 'associatedWork',
-        'gco': 'collectionDesignation',
-        'gtl': 'groupTitle',
-        'oac': 'actNumber',
-        'oco': 'commission',
-        'ocy': 'countryOfComposition',
-        'ode': 'dedication',
-        'omd': 'movementName',
-        'omv': 'movementNumber',
-        'onm': 'number',
-        'opc': 'localeOfComposition',  # origin in abc
-        'opr': 'parentTitle',
-        'ops': 'opusNumber',
-        'osc': 'sceneNumber',
-        'ota': 'alternativeTitle',
-        'otl': 'title',
-        'otp': 'popularTitle',
-        'ovm': 'volume',
-        'txl': 'textLanguage',
-        'txo': 'textOriginalLanguage',
-    }
+    # We differentiate between allUniqueNames, a list of all the values you can get,
+    # and searchAttributes, a list of all the names you can search for.  The difference
+    # is that searchAttributes has all the extra workIds in it that are different from
+    # the uniqueName for that same property.  For example, 'dedicatedTo' is a uniqueName
+    # for which the workId is 'dedication'.  Both will show up in searchAttributes (so
+    # you can search for either one), but only 'dedicatedTo' will show up in allUniqueNames,
+    # because you don't want to get both 'dedicatedTo' and 'dedication', since they are
+    # the same value.  Note that searchAttributes' initialization uses set() to get rid
+    # of any workIds that are the same name as the uniqueName.
 
     # add more as properties/import exists
-    searchAttributes = tuple(sorted([
-        'composer',
-        'copyright',
-        'date',
+    allUniqueNames = tuple(sorted([
         'fileFormat',
         'fileNumber',
         'filePath',
-    ] + list(workIdAbbreviationDict.values())))
+    ] + properties.ALL_UNIQUE_NAMES))
 
-    workIdLookupDict = {}
-    for key, value in workIdAbbreviationDict.items():
-        workIdLookupDict[value.lower()] = key
+    searchAttributes: t.Tuple = tuple(sorted(set(
+        list(allUniqueNames) + properties.ALL_MUSIC21_WORK_IDS)))
 
     # INITIALIZER #
 
     def __init__(self, *args, **keywords):
         super().__init__()
 
-        # a list of Contributor objects
-        # there can be more than one composer, or any other combination
-        self.contributors = []  # use addContributor to add.
-        self._date = None
+        self._metadata: t.Dict[str, t.List[ValueType]] = {}
+        self.software: t.List[str] = [defaults.software]
```

Comment:
-1 on having software separate -- why can't it be a metadata type like everything else?  see the Humdrum encoding type.
Also not a huge fan of having `Metadata._metadata` -- does not add any information.  `_lookup`? `_storage`?

Reply:
_contents?

Reply:
There is code out there that appends to md.software.  Is it OK to break them?

Reply:
I have renamed _metadata -> _contents.  I'll put software in _contents soon.

Reply:
Software is now in _contents.

Reply:
Done.